src/paho-mqtt.js

```diff
@@ -2323,8 +2323,13 @@ function onMessageArrived(message) {
 					get : function () {
 						if (typeof payload === "string")
 							return payload;
-						else
-							return parseUTF8(payload, 0, payload.length);
+						else {
+							try {
+								return parseUTF8(payload, 0, payload.length);
+							} catch (error) {
+								return "<" + typeof payload + ">";
```

Comment:
Why would it return response type instead of throw?

Reply:
We added this to receive binary data, to be parsed downstream in our application. Without this try-catch, the getter would simply throw an error when receiving a value not castable to a string. 
After catching the exception, we choose to return the type instead of an empty string or a null value.

Reply:
I second adding this patch. 
Subscribing to a wildcard topic means you can't be sure what types you get.
Adding this patch prevented my script from dying.

Reply:
Hi, i am facing the same issue too, wondering when this could be merged and released ?
thx !